packages/astro/src/runtime/server/render/astro/instance.ts

```diff
@@ -31,8 +31,18 @@ export class AstroComponentInstance {
 		this.factory = factory;
 		this.slotValues = {};
 		for (const name in slots) {
-			const value = slots[name](result);
-			this.slotValues[name] = () => value;
+			// prerender the slots eagerly to make collection entries propagate styles and scripts
+			let value: ReturnType<typeof slots[string]> | null = slots[name](result);
+			this.slotValues[name] = () => {
+				// use prerendered value only once
+				if (value !== null) {
+					const consumedValue = value
+					value = null
+					return consumedValue
+				}
+				// render afresh for the advanced use-case where the same slot is rendered multiple times 
+				return slots[name](result);
```

Comment:
This code definitely works, but I feel like a simple toggle might be easier to understand? Something like...

```js
let didRender = false;
let value = slots[name](result);
this.slotValues[name] = () => {
  if (!didRender) {
		didRender = true;
    return value;
  }
  return slots[name](result);
```

Reply:
Much cleaner, thanks!